### PR TITLE
Update CI to use self-hosted runner

### PR DESCRIPTION
**Pull request by nacardin**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -98,33 +98,23 @@ jobs:
...
-        run: |
-          curl -LO https://storage.googleapis.com/minikube/releases/latest/minikube-linux-amd64
-          sudo install minikube-linux-amd64 /usr/local/bin/minikube
       - name: Install Minikube on Mac
         if: startsWith(matrix.os, 'macOS')
         run:  brew install minikube
       - name: Setup Minikube for Linux
         if: startsWith(matrix.os, 'ubuntu')
         run: |  
+          minikube delete
```
Comment (simlay) [inline .github/workflows/ci.yml:117]: Are there teardowns on github CI? It seems like this would be a good thing to have as a cleanup step as well.

Comment (sehz) [reply .github/workflows/ci.yml:117]: well, new CI runs get new VM....

Comment (nacardin) [reply .github/workflows/ci.yml:117]: Good idea

**Review by sehz — APPROVED** [review]
Great.  We will finally have stable CI for integration tests!

**Comment by ajhunyady** [comment]
@nacardin this is great! The red dot in CI was pretty scary... glad to know it will be green now. ;)

**Review by sehz — APPROVED** [review]
Phew! Finally TLS passes, Great work

**Pull request merged**